src/bug-filing/github/github-bug-filing-service.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as _ from 'lodash';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import * as React from 'react';
+
+import { NamedSFC } from '../../common/react/named-sfc';
+import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { BugFilingService } from '../types/bug-filing-service';
+import { SettingsFormProps } from '../types/settings-form-props';
+
+export type GithubBugFilingSettings = {
+    repository: string;
+};
+
+function buildStoreData(repository: string): GithubBugFilingSettings {
+    return {
+        repository,
+    };
+}
+
+function isSettingsValid(data: GithubBugFilingSettings): boolean {
+    return !!data && !_.isEmpty(data.repository.trim());
+}
+
+const renderSettingsForm = NamedSFC<SettingsFormProps<GithubBugFilingSettings>>('BugFilingSettings', props => {
+    const onGitHubRepositoryChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+        props.deps.userConfigMessageCreator.setBugServiceProperty('gitHub', 'repository', newValue);
```

Comment:
This is the `key` of this service. Should we have a const in this file and use it on both places?

Reply:
I'm also worried about `repository` being pass as an string here, as this is the name of the only property of `GithubBugFilingSettings`. Any way we can tide those two value/name together?

Reply:
Yes we should. This snippet was copied from bug-filing-settings. I'll make the changes to address this issue.

Reply:
After a closer look, I think we should not pass service key in the payload. instead the store should take the selected bug service key to update its data. But for now, I'll leave it as it is.

Reply:
That's an assumption we're always updating settings for the currently selected bug filing service. Having the option to update settings for any service (selected or not) is not too expensive (we already have it) and it makes the API more flexible.